lib/active_data/model/representation.rb

```diff
@@ -67,14 +68,25 @@ def run_validations! #:nodoc:
       #
       def emerge_represented_attributes_errors!
         self.class.represented_attributes.each do |attribute|
-          key = :"#{attribute.reference}.#{attribute.column}"
-          # Rails 5 pollutes messages with an empty array on key data fetch attempt
-          messages = errors.messages[key] if errors.messages.key?(key)
-          if messages.present?
-            errors[attribute.column].concat(messages)
-            errors.delete(key)
+          move_errors(:"#{attribute.reference}.#{attribute.column}", attribute.column)
+        end
+      end
+
+      case ActiveModel.version.to_s
```

Comment:
I think it worth using Gem::Version for determining the major/minor verisons

Reply:
I had initially thought about it, but it comes with a performance penalty that I'm not sure you want to pay. When Rails 8 is out, you will hopefully be ready to get rid of all these legacy methods, or it will be a breeze to just add `, /^8\./` in a couple of places.

Reply:
Yeah, but it's a performance hit only on initialization and that's it, runtime is not affected.

Reply:
ok, I have applied it. The only drawback is that prerelases will not be handled correctly, but we are way far from the affected prereleases, it would be very weird if someone wanted to run under ruby 2.7-pre or rails 6.1-pre 🤯

Reply:
Well, it it happens :)